torchsynth/module.py

```diff
@@ -376,27 +379,23 @@ class TorchVCO(TorchSynthModule):
             name="mod_depth",
             description="depth of the pitch modulation in semitones",
         ),
-        # TODO: I think phase should go in here??
+        ModuleParameterRange(
+            -torch.pi,
+            torch.pi,
+            name="initial_phase",
+            description="Initial phase for this oscillator",
+        ),
     ]
 
     def __init__(
         self,
         synthglobals: TorchSynthGlobals,
-        phase: Optional[T] = None,
         **kwargs: Dict[str, T],
     ):
         super().__init__(synthglobals, **kwargs)
-
-        # Setup initial phase values
-        if phase is not None:
-            self.phase = nn.Parameter(data=phase.unsqueeze(1), requires_grad=False)
-        else:
-            # Create initial phase of zeros like the parameters
-            self.phase = nn.Parameter(
-                data=torch.zeros((self.batch_size,)).unsqueeze(1), requires_grad=False
-            )
-
-        assert self.phase.shape[0] == self.batch_size
+        self.phase = nn.Parameter(
+            data=self.get_parameter("initial_phase"), requires_grad=False
+        )
```

Comment:
Not sure if there is a better way to do this -- making it a parameter makes sure that it gets transferred to the GPU when the module is transferred

Reply:
hmm no don't think so

we would be tuning initial phase but not the phase if we rerun this several times is my understanding.

so this should just be a like:
self.phase = T(self.get_parameter("initial_phase"), requires_grad, device=device)

See https://towardsdatascience.com/7-tips-for-squeezing-maximum-performance-from-pytorch-ca4a40951259 Construct tensors directly on GPUs

Reply:
the problem is that the parameters haven't been transferred to the GPU by this point. if you call:
```
vco = TorchSineVCO().to(device)
```
the constructor is called and self.phase gets created on the CPU. Then all the parameters get moved onto the GPU on the `to(device)` call, but self.phase stays on the CPU, which is a problem. By making self.phase a nn.Parameter, then it also gets moved to the GPU on `to(device)`